src/achievement/achievement.rs

```diff
@@ -26,9 +72,12 @@ pub trait Rule {
     fn process(&mut self, commit: &git2::Commit, repo: &git2::Repository) -> Option<Achievement>;
 
     /// Called when finished processing all commits
-    // TODO: There may need to be some lifetimes added / specified, so that implementors of the
-    // Rule can store the commit references? But I actually think storing the commits themselves
-    // isn't quite so useful?
+    ///
+    /// This exists to enable rules that accumulate state (like calculating the shortest commit
+    /// message) to generate achievements once all commits have been visited.
+    ///
+    /// TODO: Reset Rule state in between each repository. Should this be managed by the Rules, or
+    /// should it be managed by the Rule registry and commit iterator?
```

Comment:
Too error prone to require each `Rule` implementation remember to reset its state on `finalize()` - reconstruct each rule for each repository.

But then, that gets into how the rules are created on startup, which is an area I want to experiement with `inventory`.

Maybe instead of registering `impl Rule`s with `inventory`, I can register a `const fn factory(cfg: &Config) -> impl Rule;`?